src/main/java/net/starschema/clouddb/jdbc/Oauth2Bigquery.java

```diff
@@ -376,24 +372,8 @@ public static String generateAccessToken(
     GoogleCredentials credential =
         createServiceAccountCredential(
             serviceaccountemail, keypath, password, jsonAuthContents, true);
-    HttpRequestTimeoutInitializer httpRequestInitializer =
-        new HttpRequestTimeoutInitializer(credential);
-
-    IAMCredentials.Builder builder =
-        new IAMCredentials.Builder(HTTP_TRANSPORT, JSON_FACTORY, httpRequestInitializer)
-            .setApplicationName(applicationName);
-
-    IAMCredentials iamCredentials = builder.build();
-
-    String name = "projects/-/serviceAccounts/" + serviceaccountemail;
-    GenerateAccessTokenRequest request = new GenerateAccessTokenRequest();
-    request.setScope(Collections.singletonList(BigqueryScopes.CLOUD_PLATFORM));
 
-    IAMCredentials.Projects.ServiceAccounts.GenerateAccessToken generateAccessToken;
-    generateAccessToken =
-        iamCredentials.projects().serviceAccounts().generateAccessToken(name, request);
-    GenerateAccessTokenResponse response = generateAccessToken.execute();
-    return response.getAccessToken();
+    return credential.refreshAccessToken().getTokenValue();
```

Comment:
So is `refreshAccessToken` the equivalent to making a IAMCredentials request? I wonder why we weren't using this before. Do you know if this was added in a more recent version of the credentials lib?

Reply:
Yeah, it looks like the equivalent to an IAMCredentials request. This is the [source code](https://github.com/googleapis/google-auth-library-java/blob/38bdf60189b44171f5d481fa934f4ece60553653/oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java#L497) for `refreshAccessToken`. The initial implementation was 9 years ago and the last edit was 10 months ago.

Reply:
Im wondering about the reference to `BigqueryScopes.CLOUD_PLATFORM` - I don't see the scopes being carried over in the request in `refreshAccessToken`.  This is probably fine but I don't trust the tests we have for OAuth in this repo much.